### PR TITLE
[cleanup] Fix infodict field types

### PR DESCRIPTION
**Pull request by seproDev**
**IMPORTANT**: PRs without the template will be CLOSED

### Description of your *pull request* and other information

Cleanup of various infodict field types. Requires https://github.com/yt-dlp/yt-dlp/pull/8917 to be merged.

<details open><summary>Template</summary> 

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8) and [ran relevant tests](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check all of the following options that apply:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Fix or improvement to an extractor (Make sure to add/update tests)
- [ ] New extractor ([Piracy websites will not be accepted](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy))
- [ ] Core bug fix/improvement
- [ ] New feature (It is strongly [recommended to open an issue first](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#adding-new-feature-or-making-overarching-changes))

</details>

### CODE REVIEW

File: yt_dlp/extractor/archiveorg.py [hunk 1]
```diff
@@ -167,7 +167,7 @@ class ArchiveOrgIE(InfoExtractor):
             'upload_date': '20160610',
             'description': 'md5:f70956a156645a658a0dc9513d9e78b7',
             'uploader': 'dimitrios@archive.org',
-            'creator': ['British Broadcasting Corporation', 'Time-Life Films'],
+            'creator': 'British Broadcasting Corporation, Time-Life Films',
```
Comment (pukkandan) [inline yt_dlp/extractor/archiveorg.py:170]: Should we change the field to `creators`? Ref #8917

Comment (seproDev) [reply yt_dlp/extractor/archiveorg.py:170]: It isn't currently part of that PR. But if we want to add `creators`, I can revert those changes.
I can then change this to `creators` when changing the other instances of `genre`, `artist`, `composer`, and `album_artist`.

Comment (pukkandan) [reply yt_dlp/extractor/archiveorg.py:170]: https://github.com/yt-dlp/yt-dlp/pull/8917/commits/916acca08f69bb8aca3217188915cc3a631bf3bc

You can prepare this PR to merge after that one

Comment (seproDev) [reply yt_dlp/extractor/archiveorg.py]: Done. This PR is now based on #8917 and can be merged afterwards.

File: yt_dlp/extractor/cgtn.py [hunk 2]
```diff
@@ -29,8 +29,8 @@ class CGTNIE(InfoExtractor):
                 'title': 'China, Indonesia vow to further deepen maritime cooperation',
                 'thumbnail': r're:^https?://.*\.png$',
                 'description': 'China and Indonesia vowed to upgrade their cooperation into the maritime sector and also for political security, economy, and cultural and people-to-people exchanges.',
-                'author': 'CGTN',
-                'category': 'China',
+                'creator': 'CGTN',
```
Comment (bashonly) [inline yt_dlp/extractor/cgtn.py:32]: test will fail w/o `creators`, so might as well fix that while we're at it
```suggestion
                'creators': ['CGTN'],
```

File: yt_dlp/extractor/cgtn.py [hunk 3]
```diff
@@ -46,16 +46,16 @@ def _real_extract(self, url):
 
         download_url = self._html_search_regex(r'data-video ="(?P<url>.+m3u8)"', webpage, 'download_url')
         datetime_str = self._html_search_regex(r'<span class="date">\s*(.+?)\s*</span>', webpage, 'datetime_str', fatal=False)
+        category = self._html_search_regex(r'<span class="section">\s*(.+?)\s*</span>', webpage, 'category', fatal=False)
```
Comment (bashonly) [inline yt_dlp/extractor/cgtn.py:49]:
```suggestion
        datetime_str = self._html_search_regex(
            r'<span class="date">\s*(.+?)\s*</span>', webpage, 'datetime_str', fatal=False)
        category = self._html_search_regex(
            r'<span class="section">\s*(.+?)\s*</span>', webpage, 'category', fatal=False)
        author = self._html_search_regex(
            r'<div class="news-author-name">\s*(.+?)\s*</div>', webpage, 'author', default=None),
```

File: yt_dlp/extractor/cgtn.py [hunk 4]
```diff
@@ -46,16 +46,16 @@ def _real_extract(self, url):
...
             'description': self._og_search_description(webpage, default=None),
             'thumbnail': self._og_search_thumbnail(webpage),
             'formats': self._extract_m3u8_formats(download_url, video_id, 'mp4', 'm3u8_native', m3u8_id='hls'),
-            'category': self._html_search_regex(r'<span class="section">\s*(.+?)\s*</span>',
-                                                webpage, 'category', fatal=False),
-            'author': self._html_search_regex(r'<div class="news-author-name">\s*(.+?)\s*</div>',
-                                              webpage, 'author', default=None, fatal=False),
+            'categories': [category] if category else None,
+            'creator': self._html_search_regex(r'<div class="news-author-name">\s*(.+?)\s*</div>',
+                                               webpage, 'author', default=None, fatal=False),
```
Comment (bashonly) [inline yt_dlp/extractor/cgtn.py:59]:
```suggestion
            'creators': [author] if author else None,
```

File: yt_dlp/extractor/megaphone.py [hunk 5]
```diff
@@ -8,15 +8,15 @@ class MegaphoneIE(InfoExtractor):
...
         'md5': '4816a0de523eb3e972dc0dda2c191f96',
         'info_dict': {
             'id': 'GLT9749789991',
             'ext': 'mp3',
             'title': '#97 What Kind Of Idiot Gets Phished?',
             'thumbnail': r're:^https://.*\.png.*$',
-            'duration': 1776.26375,
-            'author': 'Reply All',
+            'duration': 1998.36,
+            'creator': 'Reply All',
```
Comment (bashonly) [inline yt_dlp/extractor/megaphone.py:19]:
```suggestion
            'creators': ['Reply All'],
```

File: yt_dlp/extractor/megaphone.py [hunk 6]
```diff
@@ -40,7 +40,7 @@ def _real_extract(self, url):
             'id': video_id,
             'thumbnail': thumbnail,
             'title': title,
-            'author': author,
+            'creator': author,
```
Comment (bashonly) [inline yt_dlp/extractor/megaphone.py:43]:
```suggestion
            'creators': [author] if author else None,
```

File: yt_dlp/extractor/pladform.py [hunk 7]
```diff
@@ -35,7 +35,7 @@ class PladformIE(InfoExtractor):
             'thumbnail': str,
             'view_count': int,
             'description': str,
-            'category': list,
+            'categories': list,
```
Comment (bashonly) [inline yt_dlp/extractor/pladform.py:38]: no longer extracted, dead test, we can just remove imo
```suggestion
```

**Review by bashonly — APPROVED** [review]

**Pull request merged**